tests/test_dicts.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2013-2015 Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for :mod:`dicts`"""
+
+import unittest
+
+import mock
+import six
+
+from simpl import dicts
+
+
+class TestDicts(unittest.TestCase):
+
+    """Tests for :mod:`dicts`."""
+
+    def test_merge_dictionary(self):
+        dst = dict(
+            a=1,  # not in source
+            b=2,  # changed by source
+            c=dict(  # deep merge check
+                ca=31,
+                cc=33,
+                cd=dict(cca=1)
+            ),
+            d=4,
+            f=6,
+            g=7,
+            i=[],  # add to empty list
+            k=[3, 4],
+            l=[[], [{'s': 1}]]
+        )
+        src = dict(
+            b='u2',
+            c=dict(
+                cb='u32',
+                cd=dict(
+                    cda=dict(
+                        cdaa='u3411',
+                        cdab='u3412'
+                    )
+                )
+            ),
+            e='u5',
+            h=dict(i='u4321'),
+            i=[1],
+            j=[1, 2],
+            l=[None, [{'t': 8}]]
+        )
+        result = dicts.merge_dictionary(dst, src)
+        self.assertIsInstance(result, dict)
+        self.assertEqual(result['a'], 1)
+        self.assertEqual(result['d'], 4)
+        self.assertEqual(result['f'], 6)
+        self.assertEqual(result['b'], 'u2')
+        self.assertEqual(result['e'], 'u5')
+        self.assertIs(result['c'], dst['c'])
+        self.assertIs(result['c']['cd'], dst['c']['cd'])
+        self.assertEqual(result['c']['cd']['cda']['cdaa'], 'u3411')
+        self.assertEqual(result['c']['cd']['cda']['cdab'], 'u3412')
+        self.assertEqual(result['g'], 7)
+        self.assertIs(src['h'], result['h'])
+        self.assertEqual(result['i'], [1])
+        self.assertEqual(result['j'], [1, 2])
+        self.assertEqual(result['k'], [3, 4])
+        self.assertEqual(result['l'], [[], [{'s': 1, 't': 8}]])
+
+    def test_merge_lists(self):
+        dst = [[], [2], [None, 4]]
+        src = [[1], [], [3, None]]
+        result = dicts.merge_lists(dst, src)
+        self.assertIsInstance(result, list)
+        self.assertEqual(result[0], [1])
+        self.assertEqual(result[1], [2])
+        self.assertEqual(result[2], [3, 4], "Found: %s" % result[2])
```

Comment:
I've updated the PR with [this](https://github.com/checkmate/simpl/pull/40/files#diff-029888ac0a1327ac5d39627e785ea1ddR80) which documents a use case and explains the use of `None` values.